### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by turkycat**
Adding SetworkSerial class and also fixed a bug in UsbSerial where the program will crash when the device service is not found or accessible.

### CODE REVIEW

File: source/Serial/NetworkSerial.cpp [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+	connectToHostAsync( _host, _port )
+		.then( [ this ]( Concurrency::task<void> t )
+	{
+		try
+		{
+			t.get();
+		}
+		catch( Platform::Exception ^e )
+		{
+			ConnectionFailed( ref new Platform::String( L"BluetoothSerial::connectAsync failed with a Platform::Exception type. Message: " ) + e->Message );
```
Comment (zfields) [inline source/Serial/NetworkSerial.cpp]: BluetoothSerial:: s/b NetworkSerial::

File: source/Serial/NetworkSerial.cpp [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+using namespace Microsoft::Maker::Serial;
+
+//******************************************************************************
+//* Constructors
+//******************************************************************************
+
+NetworkSerial::NetworkSerial(
+	Windows::Networking::HostName ^host,
+	uint16_t port
```
Comment (zfields) [inline source/Serial/NetworkSerial.cpp]: `host_` and `port_` for the sake of consistency

File: source/Serial/NetworkSerial.h [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+{
+public:
+	event RemoteWiringConnectionCallback^ ConnectionEstablished;
+	event RemoteWiringConnectionCallback^ ConnectionLost;
+	event RemoteWiringConnectionFailedCallback^ ConnectionFailed;
+
+	[Windows::Foundation::Metadata::DefaultOverload]
+	NetworkSerial(
+		Windows::Networking::HostName ^host,
+		uint16_t port
```
Comment (zfields) [inline source/Serial/NetworkSerial.h]: host_  and  port_  for the sake of consistency

File: source/Serial/NetworkSerial.h [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	Windows::Storage::Streams::DataReaderLoadOperation ^_current_load_operation;
+	Windows::Storage::Streams::DataWriterStoreOperation ^_current_store_operation;
+	Windows::Networking::Sockets::StreamSocket ^_stream_socket;
+	Windows::Storage::Streams::DataReader ^_rx;
+	Windows::Storage::Streams::DataWriter ^_tx;
+
+	Concurrency::task<void>
+		connectToHostAsync(
+			Windows::Networking::HostName ^host,
+			uint16_t port
```
Comment (zfields) [inline source/Serial/NetworkSerial.h]: host_  and  port_  for the sake of consistency

File: source/Serial/NetworkSerial.cpp [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+}
+
+//******************************************************************************
+//* Private Methods
+//******************************************************************************
+
+Concurrency::task<void>
+NetworkSerial::connectToHostAsync(
+	Windows::Networking::HostName ^host,
+	uint16_t port
```
Comment (zfields) [inline source/Serial/NetworkSerial.cpp]: host_  and  port_  for the sake of consistency

File: source/Serial/NetworkSerial.cpp [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+//******************************************************************************
+//* Destructors
+//******************************************************************************
+
+NetworkSerial::~NetworkSerial(
+	void
+	)
+{
+	end();
```
Comment (zfields) [inline source/Serial/NetworkSerial.cpp]: fire off a `ConnectionLost();` here

File: source/Serial/NetworkSerial.cpp [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+	// Check to see if connection is ready
+	if( !connectionReady() ) {
+		return 0;
+	}
+	else {
+		return _rx->UnconsumedBufferLength;
+	}
+}
+
+/// \details Immediately discards the incoming parameters, because they are used for standard serial connections and will have no bearing on a bluetooth connection.
```
Comment (zfields) [inline source/Serial/NetworkSerial.cpp]: Make this comment relevant to networking

**Comment by turkycat** [comment]
Good comments. I have corrected the clarity and consistency issues.
I also cleaned up a few more potential crashes for null objects and changes the destructor's behavior with ConnectionLost.

**Pull request merged**